x-pack/plugins/lens/public/react_embeddable/renderer/lens_embeddable_component.tsx

```diff
@@ -29,18 +29,18 @@ export function LensEmbeddableComponent({
     expressionParams,
     // used for functional tests
     renderCount,
-    // has the render completed?
-    hasRendered,
     // these are blocking errors that can be shown in a badge
     // without replacing the entire panel
     blockingErrors,
+    // has the render completed?
+    hasRendered,
     // has view mode changed?
     latestViewMode,
   ] = useBatchedPublishingSubjects(
     internalApi.expressionParams$,
     internalApi.renderCount$,
-    internalApi.hasRenderCompleted$,
```

Comment:
If I'm not wrong this is the only subscription we have to `hasRenderCompleted$` if that is the case should we remove completely this observable and all the code that calls it?

Reply:
The observable is still used internally (see `dispatchError`, `dispatchRenderStart`, `dispatchRenderComplete` functions).
The subscription here is passive so a `PublishingSubject` it is ok as replacement, but for the other uses we still need a full `BehaviorSubject` like `hasRenderCompleted$`)